### ISSUE TITLE
Argument 1 passed to RainLoop\Model\Account::SetPassword() error when changing password

### ISSUE THREAD

**Issue by jerheij** [the-djmaze/snappymail#1502]
**Describe the bug**
I am running Snappymail 2.35.4 with the change password plugin 2.20. I have set up a custom DSN, user, password and statement to change the password in the MariaDB database. The changing password works but I still end up with the following error:

![image](https://github.com/the-djmaze/snappymail/assets/14292978/20cbc492-c2a1-4143-984c-726ed6cef0f5)

And the following in the logfiles:
```
Next RainLoop\Exceptions\ClientException: AuthError[102] in /usr/share/snappymail/snappymail/v/2.35.4/app/libraries/RainLoop/Actions/UserAuth.php:419
Stack trace:
#0 /usr/share/snappymail/snappymail/v/2.35.4/app/libraries/RainLoop/Actions/UserAuth.php(132): RainLoop\Actions->imapConnect()
#1 /usr/share/snappymail/snappymail/v/2.35.4/app/libraries/RainLoop/Actions/User.php(42): RainLoop\Actions->LoginProcess()
#2 /usr/share/snappymail/snappymail/v/2.35.4/app/libraries/RainLoop/ServiceActions.php(134): RainLoop\Actions->DoLogin()
#3 /usr/share/snappymail/snappymail/v/2.35.4/app/libraries/RainLoop/Service.php(132): RainLoop\ServiceActions->ServiceJson()
#4 /usr/share/snappymail/snappymail/v/2.35.4/app/libraries/RainLoop/Service.php(14): RainLoop\Service::RunResult()
#5 /usr/share/snappymail/snappymail/v/2.35.4/include.php(135): RainLoop\Service::Handle()
#6 /usr/share/snappymail/index.php(11): include('/usr/share/snap...')
#7 {main}
[2024-03-17 15:10:59.128][4b9d754b] SERVICE[WARNING]: Argument 1 passed to RainLoop\Model\Account::SetPassword() must be an instance of SnappyMail\SensitiveString, string given, called in /var/lib/snappymail/_data_/_default_/plugins/change-password/index.php on line 199
#0 /var/lib/snappymail/_data_/_default_/plugins/change-password/index.php(199): RainLoop\Model\Account->SetPassword()
#1 /usr/share/snappymail/snappymail/v/2.35.4/app/libraries/RainLoop/Plugins/Manager.php(372): ChangePasswordPlugin->ChangePassword()
#2 /usr/share/snappymail/snappymail/v/2.35.4/app/libraries/RainLoop/ServiceActions.php(137): RainLoop\Plugins\Manager->RunAdditionalJson()
#3 /usr/share/snappymail/snappymail/v/2.35.4/app/libraries/RainLoop/Service.php(132): RainLoop\ServiceActions->ServiceJson()
#4 /usr/share/snappymail/snappymail/v/2.35.4/app/libraries/RainLoop/Service.php(14): RainLoop\Service::RunResult()
#5 /usr/share/snappymail/snappymail/v/2.35.4/include.php(135): RainLoop\Service::Handle()
#6 /usr/share/snappymail/index.php(11): include('/usr/share/snap...')
#7 {main}
```

**Expected behavior**
Successful password change without error

**Please complete the following information:**
- PHP version: 7.4.3
- Change password version: 2.20
- SnappyMail Version: 2.35.4
- Mode: standalone

**Comment by the-djmaze** [comment]
`SERVICE[WARNING]: Argument 1 passed to RainLoop\Model\Account::SetPassword() must be an instance of SnappyMail\SensitiveString, string given, called in /var/lib/snappymail/_data_/_default_/plugins/change-password/index.php on line 199
`

Is fixed for next release
https://github.com/the-djmaze/snappymail/commit/8c033f53fa775c4711f6ce23f78c691ed3abb486

And an improvement:
https://github.com/the-djmaze/snappymail/commit/9ccd6161322f37c176eff2345cc1c882c54fcb22

**Comment by jerheij** [comment]
Ah okay, good to know! Thanks for your quick response! 😃

**Comment by the-djmaze** [comment]
For now you can change change-password/index.php on line 199 with:
```php
		$oAccount->SetPassword(new \SnappyMail\SensitiveString($sNewPassword));
```

**Comment by the-djmaze** [comment]
I've released 2.21 this might solve it

**Comment by jerheij** [comment]
Works! Error doesn't show any more!

Thanks again!
